File: core/src/test/java/com/larryhsiao/nyx/attachments/NewAttachmentTest.java
package com.larryhsiao.nyx.attachments;

import com.larryhsiao.nyx.core.attachments.AttachmentDb;
import com.larryhsiao.nyx.core.attachments.NewAttachment;
import com.larryhsiao.nyx.core.jots.JotsDb;
import com.silverhetch.clotho.Source;
import com.silverhetch.clotho.database.h2.MemoryH2Conn;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.sql.Connection;

/**
 * Test for {@link NewAttachment}
 */
public class NewAttachmentTest {
    /**
     * Check new attachment available.
     */
    @Test
    public void checkInsert() {
        final Source<Connection> db = new AttachmentDb(new JotsDb(new MemoryH2Conn()));
        Assertions.assertEquals(
            1L,
            new NewAttachment(
                db, "Uri", 1L
            ).value().id()
        );
    }
}

File: core/src/test/java/com/larryhsiao/nyx/old/attachments/NewAttachmentTest.kt
package com.larryhsiao.nyx.old.attachments

import com.larryhsiao.nyx.core.attachments.AttachmentDb
import com.larryhsiao.nyx.core.attachments.NewAttachment
import com.larryhsiao.nyx.core.jots.JotsDb
import com.larryhsiao.clotho.Source
import com.larryhsiao.clotho.database.h2.MemoryH2Conn
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.sql.Connection

/**
 * Test for [NewAttachment]
 */
class NewAttachmentTest {
    /**
     * Check new attachment available.
     */
    @Test
    fun checkInsert() {
        val db: Source<Connection> = AttachmentDb(JotsDb(MemoryH2Conn()))
        Assertions.assertEquals(
            1L,
            NewAttachment(
                db, "Uri", 1L
            ).value().id()
        )
    }
}